### ISSUE TITLE
Build Error on ramips/rt305x: ath9k_common.ko is missing

### ISSUE THREAD

**Issue by stsydow** [lede-project/source#191]
```
make package/kernel/mac80211/compile V=s
```
gave the following output but no other earier errors:
```
install -d -m0755 /home/st/dev/lede/bin/targets/ramips/rt305x/packages
/home/st/dev/lede/scripts/ipkg-build -c -o 0 -g 0 /home/st/dev/lede/build_dir/target-mipsel_24kec+dsp_musl-1.1.14/linux-ramips_rt305x/compat-wireless-2016-06-20/ipkg-mipsel_24kec_dsp/kmod-ath /home/st/dev/lede/bin/targets/ramips/rt305x/packages
Packaged contents of /home/st/dev/lede/build_dir/target-mipsel_24kec+dsp_musl-1.1.14/linux-ramips_rt305x/compat-wireless-2016-06-20/ipkg-mipsel_24kec_dsp/kmod-ath into /home/st/dev/lede/bin/targets/ramips/rt305x/packages/kmod-ath_4.4.14+2016-06-20-1_mipsel_24kec_dsp.ipk
rm -rf /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath
mkdir -p /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/stamp /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath
. /home/st/dev/lede/include/shell.sh; export modules=; probe_module() { mods="$1"; boot="$2"; shift 2; for mod in $mods; do mkdir -p /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules.d; echo "$mod" >> /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules.d/ath; done; if [ -e /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules.d/ath ]; then if [ "$boot" = "1" ]; then mkdir -p /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules-boot.d; ln -s ../modules.d/ath /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules-boot.d/; fi; modules="${modules:+$modules }$mods"; fi; }; add_module() { priority="$1"; mods="$2"; boot="$3"; shift 3; for mod in $mods; do mkdir -p /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules.d; echo "$mod" >> /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules.d/$priority-ath; done; if [ -e /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules.d/$priority-ath ]; then if [ "$boot" = "1" ]; then mkdir -p /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules-boot.d; ln -s ../modules.d/$priority-ath /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules-boot.d/; fi; modules="${modules:+$modules }$priority-ath"; fi; };  if [ -n "$modules" ]; then mkdir -p /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/etc/modules.d; mkdir -p /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/CONTROL; echo "#!/bin/sh" > /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/CONTROL/postinst-pkg; echo "[ -z \"\$IPKG_INSTROOT\" ] || exit 0" >> /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/CONTROL/postinst-pkg; echo ". /lib/functions.sh" >> /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/CONTROL/postinst-pkg; echo "insert_modules $modules" >> /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/CONTROL/postinst-pkg; chmod 0755 /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/CONTROL/postinst-pkg; fi
SHELL= flock /home/st/dev/lede/tmp/.root-copy.flock -c 'cp -fpR /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath/. /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/'
rm -rf /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/tmp-kmod-ath
touch /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/root-ramips/stamp/.kmod-ath_installed
if [ -f /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/pkginfo/mac80211.default.install.clean ]; then rm -f /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/pkginfo/mac80211.default.install /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/pkginfo/mac80211.default.install.clean; fi; echo "kmod-ath" >> /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/pkginfo/mac80211.default.install
mkdir -p /home/st/dev/lede/bin/targets/ramips/rt305x/packages /home/st/dev/lede/build_dir/target-mipsel_24kec+dsp_musl-1.1.14/linux-ramips_rt305x/compat-wireless-2016-06-20/ipkg-mipsel_24kec_dsp/kmod-ath9k-common/CONTROL /home/st/dev/lede/staging_dir/target-mipsel_24kec+dsp_musl-1.1.14/pkginfo
ERROR: module '/home/st/dev/lede/build_dir/target-mipsel_24kec+dsp_musl-1.1.14/linux-ramips_rt305x/compat-wireless-2016-06-20/drivers/net/wireless/ath/ath9k/ath9k_common.ko' is missing.
Makefile:1759: recipe for target '/home/st/dev/lede/bin/targets/ramips/rt305x/packages/kmod-ath9k-common_4.4.14+2016-06-20-1_mipsel_24kec_dsp.ipk' failed
make[2]: *** [/home/st/dev/lede/bin/targets/ramips/rt305x/packages/kmod-ath9k-common_4.4.14+2016-06-20-1_mipsel_24kec_dsp.ipk] Error 1
make[2]: Leaving directory '/home/st/dev/lede/package/kernel/mac80211'
package/Makefile:187: recipe for target 'package/kernel/mac80211/compile' failed
make[1]: *** [package/kernel/mac80211/compile] Error 2
make[1]: Leaving directory '/home/st/dev/lede'
```

Do you have any suggestions, where to start debugging?

**Comment by nbd168** [comment]
Did you select kmod-ath9k-common without selecting kmod-ath9k or kmod-ath9k-htc?
If that's the case, then this issue should be fixed in 78ae53ff2ff9ecbaadcf96f03ae8579ff773515e
Make sure you run make oldconfig again after updating.

**Comment by stsydow** [comment]
Thanks, it is fixed now.
No I had not set any of those manually.

PS:
I could not register for Flyspray. I just got an error.
So I posted here.
